src/contact/reducers/Reducer.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import * as ActionTypes from '../actions/ActionTypes';
+
+const INITIAL_STATE = {
+	details : null,
+	isEditable : false
+};
+
+export default function Reducer(state = INITIAL_STATE, action) {
+	switch (action.type) {
+		case ActionTypes.DISPLAY_INFO: {
+			return Object.assign({}, state, {
+				details: action.payload,
+				isEditable: action.payload.isEditable
+			});
+		}
+		case ActionTypes.ADD_EMAIL: {
+			const length = (Object.keys(state.details.otheremails).length)+1;
+			return Object.assign({}, state,  {
+				details: {
+					...state.details,
+					otheremails : {...state.details.otheremails,['em'+length]:action.payload}
+				}
+			});
+		}
+		case ActionTypes.ADD_TEL: {
+			const length = (Object.keys(state.details.otherphnos).length)+1;
```

Comment:
Why can't you make otherphnos  an array and simply push into new phones? Same applies to emails too. Ideally all of this should be done by components and here you should just dump the data.

Reply:
At first, I had done this using array. 
Then for updating values I needed keys, so used object for that. 
Can be done both ways.